app/src/main/java/com/easyfitness/MainActivity.java

```diff
@@ -364,6 +364,35 @@ protected void onStart() {
         SharedPreferences SP = PreferenceManager.getDefaultSharedPreferences(getBaseContext());
         boolean bShowMP3 = SP.getBoolean("prefShowMP3", false);
         this.showMP3Toolbar(bShowMP3);
+
+        this.checkLastBackup(SP);
+    }
+
+    private void checkLastBackup(SharedPreferences SP) {
+        int milliSecondsOfDay = 10000;//86400000; // 24 * 60 * 60 * 1000
+        int milliSecondsOfWeek = 30000;//604800000; // 7 * 24 * 60 * 60 * 1000
+        long milliSecondsOfMonth = 60000;//2419200000L; // 4 * 7 * 24 * 60 * 60 * 1000
+
+        long lastBackupUTCTime = SP.getLong("prefLastTimeBackupUTCTime", -1);
+        int prefBackupSetting = Integer.parseInt(SP.getString("defaultBackupSetting", "0"));
+        if (lastBackupUTCTime == -1) {
+            long currentTime = System.currentTimeMillis();
+            SP.edit().putLong("prefLastTimeBackupUTCTime", currentTime).apply();
```

Comment:
On fresh install, this has for effect to display a "Last backup time" in the setting fragment while no export has never been done.

Reply:
Solved by 911d9eaefa6b0ce06efe347728cdaa0ff4968c71.

We probably need to iterate on the naming of the UI elements. I am not so sure if they are ok. What do you think?